R/utils.R

```diff
@@ -18,3 +18,25 @@ is_empty <- function(x) {
 read_string <- function(path) {
   paste(readLines(path, warn = FALSE, encoding = "UTF-8"), collapse = "\n")
 }
+
+#' Collapse a vector into a human readable format for returning in messages
+#'
+#' @param vector The vector to collapse
+#' @param collapse Separator between each element after collapse
+#' @param limit Max number of items from vector to return in string
+#' @param end If length of vector over the max a string to append to the end
+#'
+#' @return The collapsed vector as a string.
+#' @keywords internal
+collapse <- function(vector, collapse = ", ", limit = 10, end = ", ...") {
```

Comment:
This is cute :)

Is it worth truncating at a total character width instead?  If you do `cumsum(nchar(vector))` and add on the appropriate separator you'll see how far out you go

Reply:
That is better, updated that too.